File: TimeTrackingClient/TimeTracking.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace TimeTrackingClient
{
    class TimeTracking
    {
        public int Id { get; set; }
        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }
        public Employee Employee { get; set; }
    }
}

File: TimeTrackingApp/src/main/java/scd/project/timetrackingapp/timeTracking/TimeTracking.java
package scd.project.timetrackingapp.timeTracking;

import jakarta.persistence.*;
import lombok.Data;
import scd.project.timetrackingapp.employee.Employee;

import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;

@Entity
@Data
public class TimeTracking implements Comparable<TimeTracking> {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private LocalDateTime checkIn;
    private LocalDateTime checkOut;

    @ManyToOne
    @JoinColumn(name = "employeeId")
    private Employee employee;

    @Override
    public int compareTo(TimeTracking o) {
        return (int)ChronoUnit.DAYS.between(this.checkIn, o.getCheckIn());
    }
}
